### PR TITLE
Plugins: Resolve Rubocop violations

### PR DESCRIPTION
**Pull request by bcoles**
Resolves many Rubocop violations.

Many of the changes were applied automatically using Rubocop, then followed up with manual review. Changes are spread over 3 commits for easier readability. Good luck to whoever wants to review this.

The remaining 87 violations are largely due to greedy `rescue Exception` error handling. The original author was too lazy to track down the expected exception types - and so am I. Others are due to `return` within arg parsing or use of string print formatters.

### CODE REVIEW

File: plugins/besecure.rb [hunk 1]
```diff
@@ -254,96 +205,146 @@ def cmd_besecure_report_download(*args)
...
-    end
 
-    def cmd_besecure_report_import(*args)
-      if args?(args, 2)
-        if !database?
-          print_error("Database not ready")
-          return ''
+      def cmd_besecure_report_download(*args)
+        if args?(args, 4)
+          req = Net::HTTP::Post.new('/json.cgi', initheader: { 'Host' => @hostname })
```
Comment (adfoster-r7) [inline plugins/besecure.rb:230]:
```suggestion
          req = Net::HTTP::Post.new('/json.cgi', { 'Host' => @hostname })
```

Crash:

``2.7.2 :021 > req = Net::HTTP::Post.new("/", initheader: { 'Host' => 'helloworld' }); http = Net::HTTP.new('127.0.0.1', 5000); http.set_debug_output($stdout); http.request(req)
Traceback (most recent call last):
	27: from /Users/user/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `<main>'
	26: from /Users/user/.rvm/gems/ruby-2.7.2/bin/ruby_executable_hooks:22:in `eval'
	25: from /Users/user/.rvm/gems/ruby-2.7.2/bin/irb:23:in `<main>'
	24: from /Users/user/.rvm/gems/ruby-2.7.2/bin/irb:23:in `load'
	23: from /Users/user/.rvm/gems/ruby-2.7.2/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
	 6: from (irb):21:in `<main>'
	 5: from (irb):21:in `new'
	 4: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/request.rb:15:in `initialize'
	 3: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/generic_request.rb:46:in `initialize'
	 2: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:16:in `initialize_http_header'
	 1: from /Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:16:in `each'
/Users/user/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/net/http/header.rb:21:in `block in initialize_http_header': undefined method `strip' for {"Host"=>"helloworld"}:Hash (NoMethodError)`
```

No crash:

```
2.7.2 :022 > req = Net::HTTP::Post.new("/", { 'Host' => 'helloworld' }); http = Net::HTTP.new('127.0.0.1', 5000); http.set_debug_output($stdout); http.request(req)
opening connection to 127.0.0.1:5000...
opened                                                                    
<- "POST / HTTP/1.1\r\nHost: helloworld\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nConnection: close\r\nContent-Length: 0\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\n"                 
<- ""
```

And headers verified via ncat:

```ncat -lnvp 5000
Ncat: Version 7.93 ( https://nmap.org/ncat )
Ncat: Listening on :::5000
Ncat: Listening on 0.0.0.0:5000
Ncat: Connection from 127.0.0.1.
Ncat: Connection from 127.0.0.1:49718.
POST / HTTP/1.1
Host: helloworld
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
Accept: */*
User-Agent: Ruby
Connection: close
Content-Length: 0
Content-Type: application/x-www-form-urlencoded


```

File: plugins/besecure.rb [hunk 2]
```diff
@@ -254,96 +205,146 @@ def cmd_besecure_report_download(*args)
...
+          if !database?
+            print_error('Database not ready')
+            return ''
+          end
+
+          tempfile = Tempfile.new('results')
 
-        print_status("Importing report to database.")
-        framework.db.import_file({:filename => tempfile})
+          res = cmd_besecure_report_download(args[0], 'nbe', File.dirname(tempfile) + '/', File.basename(tempfile))
```
Comment (adfoster-r7) [inline plugins/besecure.rb:304]: Not a blocker; `File.join`

File: plugins/thread.rb [hunk 3]
```diff
@@ -1,97 +1,71 @@
...
+      include Msf::Ui::Console::CommandDispatcher
 
-###
-#
-# This class illustrates a sample plugin.  Plugins can change the behavior of
-# the framework by adding new features, new user interface commands, or
-# through any other arbitrary means.  They are designed to have a very loose
-# definition in order to make them as useful as possible.
-#
-###
```
Comment (adfoster-r7) [inline plugins/thread.rb:15]: I think we want to leave these comments in still; I'm guessing this is meant to be a sample plugin? 🤔 

Edit: Nope, just left over cruft from the actual sample over here https://github.com/rapid7/metasploit-framework/blob/1b20db8900a5534c7e204be6e2e765ea537d2a32/plugins/sample.rb

**Review by adfoster-r7 — COMMENTED** [review]
Just the one HTTP `besecure` plugin to update then this should be good to go :+1:

File: plugins/thread.rb (hunk 3 above)
Comment (bcoles) [reply plugins/thread.rb:15]: Yeah the comments are redundant.

File: plugins/besecure.rb (hunk 1 above)
Comment (bcoles) [reply plugins/besecure.rb:230]: `initheader=` was changed to `initheader:` in 3 locations in the besure plugin. Presumably they should all be removed.

Comment (adfoster-r7) [reply plugins/besecure.rb:230]: Good call; Updating the other locations sounds good :+1:

Comment (bcoles) [reply plugins/besecure.rb:230]: Replaced

**Pull request merged**